src/Models/Link.js

```diff
@@ -0,0 +1,51 @@
+import { Schema, model } from 'dynamoose';
+
+import config from '../../config';
+
+// See: https://dynamoosejs.com/guide/Schema
+const schema = new Schema(
+  {
+    key: {
+      hashKey: true,
+      required: true,
+      type: String,
+    },
+    url: {
+      type: String,
+      required: true,
+      index: {
+        name: 'urlIndex',
+        global: true,
+      },
+    },
+    counter: {
+      type: Number,
+      default: 0,
+    },
+  },
+  // Automatically set 'createdAt' and 'updatedAt'
+  // fields on these records upon creation/update.
+  {
+    timestamps: true,
+  }
+);
+
+const options = {
+  // If we're running in development, wait for the local server to be started and
+  // automatically run a "migration" to create DynamoDB table if it doesn't exist.
+  // (We skip this on production instances since it has a performance impact).
+  create: ['development', 'test'].includes(process.env.NODE_ENV),
+  waitForActive: ['development', 'test'].includes(process.env.NODE_ENV),
+
+  // TODO: This seems to be bugged <...>. We should see if we can figure out what's
+  // going wrong here, and fix if possible so we can run "update" migrations.
+  update: false,
```

Comment:
This was driving me mental – the index would exist the first time I ran the application & then as soon as I'd reload the app, everything would explode with an `The table does not have the specified index: urlIndex` error.

After scratching my head for a while (and then adding this handy [query logger](https://github.com/DoSomething/bertly/blob/master/src/Logging/DynamoLogger.js)), I figured out that if we already have this `urlIndex` when the application starts, Dynamoose's "update" migrator will just delete it. And therefore, every other time it runs the application will either add or remove this index:

![Screen Shot 2020-05-13 at 10 53 00 AM](https://user-images.githubusercontent.com/583202/81828562-ec4cfe80-9507-11ea-88af-79c7100e01e1.png)

To keep things sane, I've disabled this option for now (so we _only_ run migrations the first time the application boots). In the relatively uncommon scenario where we change this schema, we can just delete the local database to re-migrate with a new schema.

I'll revisit this error later on & see if I can track down more details to file an issue in the library.

Reply:
Jeez that sounds like it was painful to debug 😬

Reply:
It was a spooky one!! 👻